src/SDKConnector.php

```diff
@@ -273,7 +274,11 @@ public function testConnection(KeyInterface $key = NULL) {
     try {
       // We use the original, non-decorated organization controller here.
       $oc = new OrganizationController($client);
-      $oc->load($credentials->getKeyType()->getOrganization($credentials->getKey()));
+      /* @var \Apigee\Edge\Api\Management\Entity\Organization $org */
+      $org = $oc->load($credentials->getKeyType()->getOrganization($credentials->getKey()));
+      if (empty($org->id())) {
```

Comment:
Please add a short comment about how this could occur because this is not expected behavior. Please mention the problematic API endpoint from #250

Reply:
Well the endpoint is not "invalid" if it answers with 200 to request. Basically, you could validate everywhere if the provided API endpoint is actually and Apigee Edge API endpoint and not httpbin or something like that. Precisely, you could only validate if the endpoint returned the expected API reponse (check its structure? well we should not hardcode that) because there is no pattern for valid endpoints thanks for on-prem installs. I hope you see this would be an overkill. This is a user error and it is not the PHP API client's task to handle those.
Actually, we should not take care of this here either, because the form element has a description and it is very well documented what should be the value of the API endpoint. But if you would like to add a sanity check, this is the right place to do it.

Reply:
I wrote the previous comment on Friday afternoon so let me explain what did I mean by: 

> Basically, you could validate everywhere if the provided API endpoint is actually and Apigee Edge API endpoint 

In Drupal, you only need to check _in one place_ whether the API endpoint that accepted the request actually returned an organization id (is it enough to check?) to confirm this is a valid API endpoint.
In the API client, you would need to add a sanity check to all controller classes (developer, api product, stats, MINT reports, API packeges, etc) to validate if the API endpoint is correct. That would be an overkill. In my opinion, [the examples](https://github.com/apigee/apigee-client-php#usage-examples) that the PHP API client contains should be already enough for a start.

Reply:
> Please add a short comment about how this could occur because this is not expected behavior. Please mention the problematic API endpoint from #250

Added a comment to the code:

> // Calling an invalid endpoint under some circumstances might return an
> // empty organization object, so we check if it indeed loaded an org.
> // @see https://github.com/apigee/apigee-edge-drupal/issues/250

Reply:
@mxr576 The _invalid_ endpoint pointed out in #250 is invalid in that it returns HTTP 200 and a JSON response, just not the expected JSON response (not an organization, in this case). So the SDK is returning an empty organization object, and that is why we are checking that the loaded org is valid by having a non-empty ID.
I agree that it would be an overkill to add this kind of check everywhere, which is why I've only added it to `SDKConnector::testConnection()`.

@cnovak I would also have expected `apigee/apigee-client-php` to throw an exception for this scenario.